### PR TITLE
Lesson6.2

### PR DESCRIPTION
**Pull request by andstartut**

### CODE REVIEW

File: Lesson6/src/model/Person.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        this.height = height;
+    }
+
+    private String personSurNameFilter(String surName) throws SurNameException {
+        if (surName == null || surName.isEmpty()) {
+            throw new SurNameException("You sent 'null'.");
+        }
+        String text = surName.trim().toLowerCase();
+        String textCapitalLetter = text.substring(0, 1).toUpperCase();
+        String textSmallLetters = text.substring(1).toLowerCase();
```
Comment (GlebLitvinov) [inline Lesson6/src/model/Person.java:32]: it is already lower case (text = surName.trim().toLowerCase();)

File: Lesson6/src/model/Person.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        String textCommon = textCapitalLetter + textSmallLetters;
+        if (textCommon.indexOf(" ") > 0) {
+            try {
+                throw new SurNameException("Wrong input " + textCommon + ". Don't use 'space'.");
+            } catch (SurNameException e) {
```
Comment (GlebLitvinov) [inline Lesson6/src/model/Person.java:37]: you don't want to catch it in the constructor, let code that creates person handle it

File: Lesson6/src/service/MilitaryBase.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    public List<Recruit> getRecruits() {
+        return recruits;
+    }
+
+    public void getSurnamesInAlphabeticalOrder() {
+        recruits.sort(Comparator.comparing(Recruit::getSurname));
+        for (Recruit recruit : recruits) {
+            System.out.println(recruit.getSurname() + ", " + recruit.getName());
```
Comment (GlebLitvinov) [inline Lesson6/src/service/MilitaryBase.java:90]: you also need to print index so the list would look like
1. Ivan Ivanov
2. Jack Jackson
...

**Review by GlebLitvinov — CHANGES_REQUESTED** [review]
please remove robot related commits from here

File: Lesson6/src/model/Person.java (hunk 2 above)
Comment (andstartut) [reply Lesson6/src/model/Person.java:37]: I did this in the Main class, when I created a person. However, when the exception  was caught the for loop was broke and person wasn't created surname. After that he wasn't added to the healthyPeople list.

File: Lesson6/src/model/Person.java (hunk 1 above)
Comment (andstartut) [reply Lesson6/src/model/Person.java:32]: OK

**Pull request closed without merging**